suite-native/module-home/src/components/BiometricsBottomSheet.tsx

```diff
@@ -0,0 +1,176 @@
+
+const cardStyle = prepareNativeStyle(utils => ({
+    alignItems: 'center',
+    textAlign: 'center',
+    borderRadius: utils.borders.radii.medium,
+    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+    paddingVertical: utils.spacings.large,
+    marginBottom: utils.spacings.large,
+    marginTop: utils.spacings.extraLarge,
+    gap: utils.spacings.large,
+}));
+
+const imageWrapperStyle = prepareNativeStyle(utils => ({
+    padding: 12,
+    borderRadius: utils.borders.radii.round,
+    backgroundColor: utils.colors.backgroundSurfaceElevation2,
+    color: utils.colors.iconPrimaryDefault,
+    width: 88,
+    height: 88,
+}));
+
+const textContentStyle = prepareNativeStyle(utils => ({
+    gap: utils.spacings.small,
+}));
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    gap: utils.spacings.medium,
+    paddingHorizontal: utils.spacings.small,
+}));
+
+type BiometricsBottomSheetProps = {
+    onClose?: (isVisible: boolean) => void;
+};
+
+export const BiometricsBottomSheet = ({
+    onClose = (_: boolean) => {},
+}: BiometricsBottomSheetProps) => {
+    const { applyStyle } = useNativeStyles();
+    const { translate } = useTranslate();
+    const { showToast } = useToast();
+    const hasDecidedBiometrics = useSelector(selectHasDecidedBiometrics);
+    const { isBiometricsOptionEnabled } = useIsBiometricsEnabled();
+
+    const [isVisible, setIsVisible] = useState(false);
+    const [biometricsTypes, setBiometricsTypes] = useState<
+        ('face' | 'fingerprint' | 'iris' | 'unknown')[]
+    >([]);
+
+    const isFace = biometricsTypes.filter(x => ['face', 'iris'].includes(x));
+
+    const getIconName = () => {
+        if (isFace) {
+            return 'faceId';
+        }
+
+        return 'fingerprint';
+    };
+
+    const getTitle = () => {
+        if (biometricsTypes.length === 0 || biometricsTypes.includes('unknown')) {
+            return <Translation id="moduleHome.biometricsModal.title.unknown" />;
+        }
+        if (isFace) {
+            return <Translation id="moduleHome.biometricsModal.title.faceId" />;
+        }
+        if (Platform.OS === 'android') {
+            return <Translation id="moduleHome.biometricsModal.title.fingerprint" />;
+        }
+        return <Translation id="moduleHome.biometricsModal.title.touchId" />;
+    };
+
+    useEffect(() => {
+        const getSupportedTypes = async () => {
+            const biometricsTypesAvailable = await getSupportedBiometricsTypes();
+            if (biometricsTypesAvailable.length === 0) {
+                return;
+            }
+            setBiometricsTypes(biometricsTypesAvailable);
+        };
+        getSupportedTypes();
+    }, []);
+
+    useEffect(() => {
+        const checkBiometrics = async () => {
+            const isBiometricsAvailable = await getIsBiometricsFeatureAvailable();
+
+            if (isBiometricsAvailable && !(isBiometricsOptionEnabled || hasDecidedBiometrics)) {
+                setTimeout(() => {
```

Comment:
Always use cleanup function for `effect` especially when dealing with timeout and intervals

Reply:
fixed in bee7c8f87